src/ui/public/vislib/lib/tilemap_settings.js

```diff
@@ -0,0 +1,229 @@
+import uiModules from 'ui/modules';
+import _ from 'lodash';
+import url from 'url';
+import marked from 'marked';
+import uiRoutes from 'ui/routes';
+
+marked.setOptions({
+  gfm: true, // Github-flavored markdown
+  sanitize: true // Sanitize HTML tags
+});
+
+/**
+ * Object to read out the map-service configuration
+ * Clients can register custom ways of resolving the url and the options
+ * @type {Object}
+ */
+const mapSettings = {
+  _lockDown: false,
+  _settingsDelegates: [],
+  _manifest: (new Promise((resolve, reject) => {
+    uiRoutes.addSetupWork(async() => {
```

Comment:
Unfortunately `async` and `await` are not compatible with Angular promises so we should avoid using it on the front-end for now